src/stream/src/executor/mview/materialize.rs

```diff
@@ -68,7 +69,10 @@ fn get_op_consistency_level(
     may_have_downstream: bool,
     depended_subscriptions: &HashSet<u32>,
 ) -> StateTableOpConsistencyLevel {
-    if !depended_subscriptions.is_empty() {
+    if !enable_strict_consistency() {
+        // disable sanity check in non-strict mode
```

Comment:
I saw that non-strict mode can be enabled via streaming config. When do we use it? Can user enable it in production?

Reply:
When users don't care about strict consistency, but care about the stream keeping running without panic, they can set the config item to `false`. This PR is exactly the last required piece to publicize this feature to users.

Reply:
The current subscription, and in the future partial checkpoint, depend on setting the consistency level to `EnableLogStore`, and otherwise some assertion may fail. If this PR aims to let storage side disable the old value check, we'd better either set the config in storage config and rewrite the definition of `ENABLE_SANITY_CHECK`, or disable subscription and partial checkpoint on meta side. Personally I prefer updating `ENABLE_SANITY_CHECK`, so that it won't be too hacky on meta side.

Reply:
Latest commit changed to the overwriting `ENABLE_SANITY_CHECK` approach, looks clean.